jupyter_ydoc/ydoc.py

```diff
@@ -10,87 +10,185 @@
 
 class YBaseDoc(ABC):
     """
-    Base class
+    Base YDoc class.
+    This class, defines the minimum API that any documents must provide
+    for the :class:`YDocWebSocketHandler` to be able to get and set the
+    content of the document as well as subscribe to changes in the document.
     """
 
     def __init__(self, ydoc: Y.YDoc):
+        """
+        Construct a YBaseDoc.
+
+        :param ydoc: The :class:`y_py.YDoc` that will hold the data of the document.
+        :type ydoc: :class:`y_py.YDoc`
+        """
         self._ydoc = ydoc
         self._ystate = self._ydoc.get_map("state")
         self._subscriptions = {}
 
     @property
-    def ystate(self):
+    def ystate(self) -> Y.YMap:
+        """
+        A :class:`y_py.YMap` containing the state of the document.
+
+        :return: That contains document's state.
+        :rtype: :class:`y_py.YMap`
+        """
         return self._ystate
 
     @property
-    def ydoc(self):
+    def ydoc(self) -> Y.YDoc:
+        """
+        The underlying :class:`y_py.YDoc` that contains the data.
+
+        :return: That contains document's data.
+        :rtype: :class:`y_py.YDoc`
+        """
         return self._ydoc
 
     @property
     def source(self):
+        """
+        Returns the content of the document.
+
+        :return: The content of the document.
+        :rtype: Any
+        """
         return self.get()
 
     @source.setter
     def source(self, value):
+        """
+        Sets the content of the document.
+
+        :param value: The content of the document.
+        :type value: Any
+        """
         return self.set(value)
 
     @property
-    def dirty(self) -> None:
+    def dirty(self) -> Optional[bool]:
+        """
+        Returns whether the document in memory has changes that are not on disk.
```

Comment:
I don't think we should mention "disk", as `jupyter_ydoc` doesn't know how the dirty flag will be handled.

Reply:
What should we say?

Reply:
Maybe:
```suggestion
        Returns whether the document has uncommitted changes.
```
I don't know if "uncommitted" is generic enough. Otherwise: `Returns whether the document is dirty`.